### PR TITLE
Unify naming between notebooks and test cases

### PR DESCRIPTION
**Pull request by romanlutz**
## Description

This PR aligns naming between notebooks and test cases. It's a prereq for the e2e-with-flighting tests I've written in #1467. This is another change refactored out of #1467 to make that PR more digestable. What does that mean in detail?
- `ClassificationModelDebugging` -> `CensusClassificationModelDebugging`
- `*modelStatistics` -> `*modelOverview`
- `casualAnalysis` -> `causalAnalysis` (love the typo, though! 😆 )

## Checklist

- [ ] I have added screenshots above for all UI changes.
- [ ] I have added e2e tests for all UI changes.
- [ ] Documentation was updated if it was needed.

### CODE REVIEW

**Comment by imatiach-msft** [comment]
"casualAnalysis -> causalAnalysis (love the typo, though! 😆 )"
hehe very funny typo indeed 😆, I guess casual analysis is when you just sometimes want to do some spontaneous analysis but aren't really serious about it?? But we are very serious about our analyses in our dashboards 🥽🥼, perhaps we should call it a "serious analysis" instead.

File: libs/e2e/src/index.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ export * from "./lib/describer/modelAssessment/causalAnalysis/describeCausalAnal
 export * from "./lib/describer/modelAssessment/dataExplorer/describeDatasetExplorer";
 export * from "./lib/describer/modelAssessment/errorAnalysis/describeErrorAnalysis";
 export * from "./lib/describer/modelAssessment/featureImportances/individualFeatureImportance/describeIndividualFeatureImportance";
-export * from "./lib/describer/modelAssessment/modelStatistics/describeModelStatistics";
+export * from "./lib/describer/modelAssessment/modelOverview/describeModelOverview";
```
Comment (vinuthakaranth) [inline libs/e2e/src/index.ts:9]: just a fyi, this has to be updated in AML once we release and this change is being used in AML.

**Review by vinuthakaranth — APPROVED** [review]

**Comment by imatiach-msft** [comment]
even in the econml docs they seem to make the same spelling mistakes:
https://econml.azurewebsites.net/spec/estimation/deepiv.html
"What is required is a vector of instruments Z, assumed to **casually** affect the distribution of the treatment T"

**Review by gaugup — APPROVED** [review]

File: libs/e2e/src/index.ts (hunk 1 above)
Comment (romanlutz) [reply libs/e2e/src/index.ts:9]: Good point! Need to keep that on our radar for when we do that. Or do you think I should proactively go and upgrade the version Studio uses and make this change?

Comment (vinuthakaranth) [reply libs/e2e/src/index.ts:9]: You can update this once we release new version with this code change, that will be fine.

**Pull request merged**